src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ChunkedEncodingWriteStream.cs

```diff
@@ -19,6 +20,8 @@ public ChunkedEncodingWriteStream(HttpConnection connection) : base(connection)
 
             public override Task WriteAsync(ReadOnlyMemory<byte> source, CancellationToken ignored)
             {
+                Debug.Assert(_connection._currentRequest != null);
```

Comment:
Is _currentRequest actually used anymore, aside from asserts?

Reply:
Just asserts and logging now. Although we could potentially use it in places to avoid passing the instance around and storing it in various async state machines.